### PR TITLE
bump kind version

### PR DESCRIPTION
**Pull request by aojea**
**- What this PR does and why is it needed**

kind latest versions contains fixes that will help to
reduce flakiness.
Last stable version uses its own docker network.

**- Special notes for reviewers**

This will simplify IPv6 and Dual stack testing too

**- How to verify it**
CI should work :)

**- Description for the changelog**

### CODE REVIEW

**Comment by trozet** [comment]
looks like the DNS cases are failing?
 [Fail] [sig-network] DNS [It] should provide DNS for pods for Hostname [LinuxOnly] [Conformance] 
/home/runner/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/dns_common.go:538

[Fail] [sig-network] DNS [It] should provide DNS for the cluster  [Conformance] 
/home/runner/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/dns_common.go:538

**Comment by aojea** [comment]
> looks like the DNS cases are failing?
> [Fail] [sig-network] DNS [It] should provide DNS for pods for Hostname [LinuxOnly] [Conformance]
> /home/runner/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/dns_common.go:538
> 
> [Fail] [sig-network] DNS [It] should provide DNS for the cluster [Conformance]
> /home/runner/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/dns_common.go:538

yep, with shared gateway, we were discussing it, it's related to the way docker implement custom networks ... working on it

**Comment by aojea** [comment]
@trozet PATL, this will simplify @Billy99 's PR for IPv6

File: contrib/kind.sh [hunk 1]
```diff
@@ -234,6 +234,32 @@ kubectl get secrets -o jsonpath='{.items[].data.token}' > /tmp/kind/token
 pushd ../go-controller
 make
 popd
+
+# Patch CoreDNS to work in Github CI
```
Comment (trozet) [inline contrib/kind.sh:238]: should this patching only be done for a Github run and not in other users' deployments?

Comment (aojea) [reply contrib/kind.sh:238]: yeah

File: contrib/kind.sh [hunk 2]
```diff
@@ -234,6 +234,34 @@ kubectl get secrets -o jsonpath='{.items[].data.token}' > /tmp/kind/token
 pushd ../go-controller
 make
 popd
+
+if [ "${GITHUB_ACTIONS:-false}" == "true" ]; then
```
Comment (aojea) [inline contrib/kind.sh:238]: https://docs.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables

> GITHUB_ACTIONS	Always set to true when GitHub Actions is running the workflow. You can use this variable to differentiate when tests are being run locally or by GitHub Actions.

**Comment by aojea** [comment]
/retest
There is 1 test failure unrelated to the change @trozet

**Comment by trozet** [comment]
@aojea I see in one run coredns failed to become ready:
2020-07-16T16:39:45.2440338Z Jul 16 16:39:45.243: INFO: The status of Pod coredns-66bff467f8-j5t2k is Running (Ready = false), waiting for it to be either Running (with Ready = true) or Failed
2020-07-16T16:39:45.2445882Z Jul 16 16:39:45.243: INFO: 13 / 14 pods in namespace 'kube-system' are running and ready (14 seconds elapsed)

Probably not due to this patch but just documenting it in case we see it consistently start happening in other CI runs.

**Comment by aojea** [comment]
yes, we should start tracking this recurrent issues,

**Pull request merged**